### PR TITLE
Fix counters read from the rom

### PR DESCRIPTION
**Pull request by Nekaida**
These are max indices in zero-based arrays, so we have to increment them by 1.

### CODE REVIEW

File: src/System/Configuration.cpp [hunk 1]
```diff
@@ -165,9 +165,11 @@ namespace ame
...
         FETCH(MapNameCount, rom.readByte());
         FETCH(MapNameTotal, rom.readByte());
-        OverworldCount++; // value is one less
-        ItemCount *= 2;
-        MapNameCount += (RomType == RT_FRLG ? 1 : 0); // correct FRLG
+        // the values from the ROM are the max indices, so we have to add 1
+        PokemonCount++;
+        OverworldCount++;
+        ItemCount = ItemCount * 2 + 1;
+        MapNameCount++;
```
Comment (Sierraffinity) [inline src/System/Configuration.cpp]: Why did you change this line? FRLG need this while RSE don't.

Comment (Nekaida) [reply src/System/Configuration.cpp]: I knew that it's the same for Emerald, but your right in that RS differs. I updated the commit.
